src/lib/y2firewall/ui_state.rb

```diff
@@ -0,0 +1,155 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Firewall
+  # Singleton class to keep the position of the user in the UI and other similar
+  # information that needs to be rememberd across UI redraws to give the user a
+  # sense of continuity.
+  class UIState
+    include Yast::I18n
+
+    # Constructor
+    #
+    # Called through {.create_instance}, starts with a blank situation (which
+    # means default for each widget will be honored).
+    def initialize
+      textdomain "firewall"
+      @candidate_nodes = []
+    end
+
+    # Method to be called when the user decides to visit a given page by
+    # clicking in one node of the general tree.
+    #
+    # It remembers the decision so the user is taken back to a sensible point of
+    # the tree (very often the last he decided to visit) after redrawing.
+    #
+    # @param [CWM::Page] page associated to the tree node
+    def go_to_tree_node(page)
+      self.candidate_nodes =
+        if page.respond_to?(:zone)
+          zone_page_candidates(page)
+        else
+          [page.label]
+        end
+      # Landing in a new node, so invalidate previous details about position
+      # within a node, they no longer apply
+      self.tab = nil
+    end
+
+    # Method to be called when the user switches to a tab within a tree node.
+    #
+    # It remembers the decision so the same tab is showed in case the user stays
+    # in the same node after redrawing.
+    #
+    # @param [CWM::Page] page associated to the tab
+    def switch_to_tab(page)
+      self.tab = page.label
+    end
+
+    # Method to be called when the user operates in a row of a table of zones
+    # or creates a new zone.
+    #
+    # @param zone [String] zone name
+    def select_row(zone_name)
```

Comment:
As we were discussing we should try to generalize the current. But for now we should at least fix the comments and param name as it will used not only for zones but also for interfaces.

Reply:
Just a leftover. Anyway, I will try to extract it.